python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -918,3 +942,218 @@ def test_validate_load_balancer_managed_outbound_ip_count(test_description, name
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret="asdfghjkl"),
+        InvalidArgumentValueError, 'Must not specify --client-secret when --enable-managed-identity is True'
+    ),
+    (
+        "Should raise InvalidArgumentValueError when version is not present",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret=None,
+             version=None),
+        InvalidArgumentValueError, 'Enabling managed identity requires --version >= 4.14.z'
+    ),
+    (
+        "Should raise InvalidArgumentValueError when version is invalid",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret=None,
+             version="a"),
+        InvalidArgumentValueError, 'Enabling managed identity requires --version >= 4.14.z'
+    ),
+    (
+        "Should raise InvalidArgumentValueError when version <= 4.14.0",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret=None,
+             version="4.13.99"),
+        InvalidArgumentValueError, 'Enabling managed identity requires --version >= 4.14.z'
+    ),
+    (
+        "Should raise RequiredArgumentMissingError when no platform workload identities are set",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret=None,
+             version="4.14.0",
+             platform_workload_identities=[]),
+        RequiredArgumentMissingError, 'Enabling managed identity requires platform workload identities to be provided'
+    ),
+    (
+        "Should raise RequiredArgumentMissingError when cluster identity is not set",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret=None,
+             version="4.14.0",
+             platform_workload_identities=[Mock(resource_id="foo")],
+             mi_user_assigned=None),
+        RequiredArgumentMissingError, 'Enabling managed identity requires cluster identity to be provided'
+    ),
+    (
+        "Should not raise any exception when valid",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret=None,
+             version="4.14.0",
+             platform_workload_identities=[Mock(resource_id="foo")],
+             mi_user_assigned="foo"),
+        None, None
+    )
+]
+
+
+@pytest.mark.parametrize(
+    "test_description, namespace, expected_exception, expected_exception_message",
+    test_validate_enable_managed_identity_data,
+    ids=[i[0] for i in test_validate_enable_managed_identity_data]
+)
+def test_validate_enable_managed_identity(test_description, namespace, expected_exception, expected_exception_message):
+    if expected_exception is None:
+        validate_enable_managed_identity(namespace)
+    else:
+        with pytest.raises(expected_exception, match=expected_exception_message):
+            validate_enable_managed_identity(namespace)
+
+
+test_validate_platform_workload_identities_data = [
```

Comment:
This list could use a `Should not raise any exception when valid` case where the identities are passed as resource IDs.